### PR TITLE
feat: terminal.group support

### PR DESCRIPTION
**Pull request by EstaniPP**
This PR adds group and groupEnd functionality to the plugin. The message is formatted on the browser, that's why we don't need to send the window GUID or sthing similar.

Small fix on queueLog file after ! addition.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -111,22 +112,40 @@ export default terminal
...
-        fetch(`/__terminal/${type}?m=${encodeURI(message)}&t=${Date.now()}&q=${queueOrder++}`)
+        fetch(`/__terminal/${type}?m=${encodeURI(message)}&t=${Date.now()}&q=${queueOrder++}&gl=${groupLevel}`)
+        break
+      }
+      case 'group': {
+        groupLevel++
+        break
+      }
+      case 'groupEnd': {
+        groupLevel = groupLevel === 0 ? groupLevel : --groupLevel
```
Comment (patak-dev) [inline src/index.ts:140]:
```suggestion
        groupLevel && --groupLevel
```

**Comment by patak-dev** [comment]
I think we should move the indent to the server for all logs. Since you can't do it for tables in the client, it feels more consistent that we do it in only one place.
It is also more flexible because later we may choose to show the grouping in a different way in the terminal compared to the console. In the console there are arrows that help you identify the grouping, here in the console we may want to add coloring, or something else.

**Comment by EstaniPP** [comment]
Yes, I agree, will change group text to the plugin, it also allow us to avoid grouping text en each method before sending it to the plugin.

I didn't add colors or arrows because each browser shows the grouping in different ways so I decided to avoid special logging.

Thanks

File: src/index.ts (hunk 1 above)
Comment (EstaniPP) [reply src/index.ts:140]: Great suggestion, will add it 👍

**Comment by EstaniPP** [comment]
Requesteed changes made @patak-dev, I put group-text outside pluginTerminal just to make it more readable, are you agree?

Regards

**Review by patak-dev — APPROVED** [review]

**Comment by patak-dev** [comment]
Awesome! Thanks a lot for all these PRs 🎉

**Pull request merged**